test/unit/gateways/shift4_test.rb

```diff
@@ -127,6 +127,60 @@ def test_successful_purchase_with_stored_credential
     end.respond_with(successful_purchase_response)
   end
 
+  def test_successful_purchase_with_card_on_file_fields
+    card_on_file_fields = {
+      usage_indicator: '01',
+      indicator: '02',
+      scheduled_indicator: '01'
+    }
+    stub_comms do
+      @gateway.purchase(@amount, @credit_card, @options.merge(card_on_file_fields))
+    end.check_request do |_endpoint, data, _headers|
+      request = JSON.parse(data)['transaction']
+      assert_equal request['cardOnFile']['usageIndicator'], card_on_file_fields[:usage_indicator]
+      assert_equal request['cardOnFile']['indicator'], card_on_file_fields[:indicator]
+      assert_equal request['cardOnFile']['scheduledIndicator'], card_on_file_fields[:scheduled_indicator]
+      assert_nil request['cardOnFile']['transactionId']
+    end.respond_with(successful_purchase_response)
+
+    card_on_file_fields = {
+      usage_indicator: '02',
+      indicator: '01',
+      scheduled_indicator: '02',
+      transaction_id: 'TXID00001293'
+    }
+    stub_comms do
+      @gateway.purchase(@amount, @credit_card, @options.merge(card_on_file_fields))
+    end.check_request do |_endpoint, data, _headers|
+      request = JSON.parse(data)['transaction']
+      assert_equal request['cardOnFile']['usageIndicator'], card_on_file_fields[:usage_indicator]
+      assert_equal request['cardOnFile']['indicator'], card_on_file_fields[:indicator]
+      assert_equal request['cardOnFile']['scheduledIndicator'], card_on_file_fields[:scheduled_indicator]
+      assert_equal request['cardOnFile']['transactionId'], card_on_file_fields[:transaction_id]
+    end.respond_with(successful_purchase_response)
+  end
+
+  def test_card_on_file_fields_and_stored_credential_framework_combined
+    card_on_file_fields = {
+      usage_indicator: '02',
+      indicator: '02',
+      scheduled_indicator: '02'
+    }
+    stored_credential_options = {
+      initial_transaction: true,
+      reason_type: 'recurring'
+    }
+    stub_comms do
+      @gateway.purchase(@amount, @credit_card, @options.merge(card_on_file_fields, { stored_credential: stored_credential_options }))
```

Comment:
I'm getting an error here, `Error: test_card_on_file_fields_and_stored_credential_framework_combined(Shift4Test): ArgumentError: wrong number of arguments (given 2, expected 1)`

Reply:
<img width="570" alt="image" src="https://user-images.githubusercontent.com/73054822/189590716-f607866a-6ec1-436a-a2de-f4e11f3308fd.png">
They are getting passed for me. I think there is an issue with merge as it doesn't support multiple arguments in older version of ruby.